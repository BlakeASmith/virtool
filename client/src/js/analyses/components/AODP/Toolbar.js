import { toNumber } from "lodash-es";
import React, { useCallback } from "react";
import { connect } from "react-redux";
import styled from "styled-components";
import { Input, SearchInput } from "../../../base";
import { setAODPFilter, setSearchIds } from "../../actions";
import { getFuse } from "../../selectors";

const StyledAODPToolBar = styled.div`
    display: grid;
    grid-template-columns: 3fr 1fr;
    grid-gap: 5px;
`;

export const AODPToolBar = ({ filterAODP, fuse, id, onSearch, onSetFilter }) => {
    const handleChangeSearch = useCallback(
        e => {
            onSearch(e.target.value, fuse);
        },
        [id]
    );

    const handleChangeFilter = e => {
        onSetFilter(toNumber(e.target.value));
    };

    return (
        <StyledAODPToolBar>
            <SearchInput onChange={handleChangeSearch} onKeyDown={e => e.stopPropagation()} placeholder="Name" />
            <Input
                type="number"
                placeholder="Minimum Identity"
                min={0.5}
                max={1.0}
                step={0.01}
                value={filterAODP.toFixed(2)}
                onChange={e => handleChangeFilter(e)}
            />
        </StyledAODPToolBar>
    );
};

const mapStateToProps = state => ({
    filterAODP: state.analyses.filterAODP,
    fuse: getFuse(state),
    id: state.analyses.detail.id
});

const mapDispatchToProps = dispatch => ({
    onSearch: (term, fuse) => {
        dispatch(setSearchIds(term ? fuse.search(term) : null));
    },

    onSetFilter: identity => {
        dispatch(setAODPFilter(identity));
    }
});

export default connect(mapStateToProps, mapDispatchToProps)(AODPToolBar);
