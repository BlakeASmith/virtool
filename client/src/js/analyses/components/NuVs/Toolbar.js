import React, { useCallback } from "react";
import { connect } from "react-redux";
import styled from "styled-components";
import { DropdownButton, FormControl, FormGroup, InputGroup, MenuItem } from "react-bootstrap";
import { LinkContainer } from "react-router-bootstrap";
import { Button, Icon, Toolbar } from "../../../base";
import { setAnalysisSortKey, setSearchIds, toggleFilterORFs, toggleFilterSequences } from "../../actions";
import { getFuse, getResults } from "../../selectors";

const StyledNuVsToolbar = styled(Toolbar)`
    margin-bottom: 10px;
`;

const sortTitles = {
    coverage: "Coverage",
    depth: "Depth",
    e: "E-Value",
    length: "Length",
    orfs: "ORFs",
    weight: "Weight"
};

const SortDropdownButtonTitle = ({ sortKey }) => (
    <span>
        <span>
            <Icon name="sort" /> Sort:{" "}
        </span>
        <span>{sortTitles[sortKey]}</span>
    </span>
);

const StyledSortDropdownButton = styled(DropdownButton)`
    align-items: center;
    display: flex;
    justify-content: space-between;
    width: 132px;
`;

const SortDropdownButton = ({ sortKey, onSelect }) => (
    <StyledSortDropdownButton
        id="nuvs-sort-dropdown"
        key={sortKey}
        title={<SortDropdownButtonTitle sortKey={sortKey} />}
        onSelect={onSelect}
    >
        <MenuItem eventKey="length">Length</MenuItem>
        <MenuItem eventKey="e">E-Value</MenuItem>
        <MenuItem eventKey="orfs">ORFs</MenuItem>
    </StyledSortDropdownButton>
);

const NuVsToolbar = ({
    filterORFs,
    filterSequences,
    fuse,
    id,
    sortKey,
    onFilterSequences,
    onFilterORFs,
    onSearch,
    onSelect
}) => {
    const handleChange = useCallback(
        e => {
            onSearch(e.target.value, fuse);
        },
        [id]
    );

    return (
        <StyledNuVsToolbar>
            <FormGroup>
                <InputGroup>
                    <InputGroup.Addon>
                        <Icon name="search" />
                    </InputGroup.Addon>
                    <FormControl
                        onChange={handleChange}
                        onKeyDown={e => e.stopPropagation()}
                        placeholder="Name or family"
                    />
                </InputGroup>
            </FormGroup>
            <SortDropdownButton sortKey={sortKey} onSelect={onSelect} />
            <Button
                icon="filter"
                onClick={onFilterSequences}
                active={filterSequences}
                tip="Hide sequences that have no HMM hits"
            >
                Filter Sequences
            </Button>
            <Button icon="filter" onClick={onFilterORFs} active={filterORFs} tip="Hide ORFs that have no HMM hits">
                Filter ORFs
            </Button>
            <LinkContainer to={{ state: { export: true } }}>
                <Button tip="Export" icon="download" />
            </LinkContainer>
        </StyledNuVsToolbar>
    );
};

const mapStateToProps = state => {
    const { detail, filterORFs, filterSequences, sortKey } = state.analyses;
    return {
        id: detail.id,
        fuse: getFuse(state),
        results: getResults(state),
        filterORFs,
        filterSequences,
        sortKey
    };
};

const mapDispatchToProps = dispatch => ({
    onFilterSequences: () => {
        dispatch(toggleFilterSequences());
    },
    onFilterORFs: () => {
        dispatch(toggleFilterORFs());
    },
    onSearch: (term, fuse) => {
        dispatch(setSearchIds(term ? fuse.search(term) : null));
    },
    onSelect: sortKey => {
        dispatch(setAnalysisSortKey(sortKey));
    }
});

export default connect(mapStateToProps, mapDispatchToProps)(NuVsToolbar);
