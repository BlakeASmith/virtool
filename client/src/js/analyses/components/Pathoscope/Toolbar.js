import React from "react";
import styled from "styled-components";
import { DropdownButton, FormControl, FormGroup, InputGroup, MenuItem } from "react-bootstrap";
import { connect } from "react-redux";
import { Button, Checkbox, Icon, Toolbar, DropDownItem, ButtonDropDown } from "../../../base";
import {
    collapseAnalysis,
    setAnalysisSortKey,
    setPathoscopeFilter,
    togglePathoscopeSortDescending,
    toggleShowPathoscopeReads
} from "../../actions";

export const PathoscopeDownloadDropdownTitle = () => (
    <span>
        <Icon name="file-download" /> Export
    </span>
);

const StyledPathoscopeToolbar = styled(Toolbar)`
    display: flex;
    margin-bottom: 10px !important;
`;

export const PathoscopeToolbar = ({
    analysisId,
    filterIsolates,
    filterOTUs,
    showReads,
    sortDescending,
    sortKey,
    onCollapse,
    onFilter,
    onSetSortKey,
    onToggleShowReads,
    onToggleSortDescending
}) => {
    return (
        <StyledPathoscopeToolbar>
            <FormGroup>
                <InputGroup>
                    <InputGroup.Button>
                        <Button title="Sort Direction" onClick={onToggleSortDescending} tip="Sort List">
                            <Icon name={sortDescending ? "sort-amount-down" : "sort-amount-up"} />
                        </Button>
                    </InputGroup.Button>
                    <FormControl componentClass="select" value={sortKey} onChange={e => onSetSortKey(e.target.value)}>
                        <option className="text-primary" value="coverage">
                            Coverage
                        </option>
                        <option className="text-success" value="pi">
                            Weight
                        </option>
                        <option className="text-danger" value="depth">
                            Depth
                        </option>
                    </FormControl>
                </InputGroup>
            </FormGroup>

            <Button
                icon="compress"
                title="Collapse"
                tip="Collapse"
                onClick={onCollapse}
                className="hidden-xs"
                disabled={false}
            />

            <Button
                icon="chart-pie"
                title="Weight Format"
                tip="Weight Format"
                active={!showReads}
                className="hidden-xs"
                onClick={onToggleShowReads}
            />

            <Button title="Filter" tip="Filter Results" onClick={onFilter} active={filterOTUs || filterIsolates}>
                <Icon name="filter" />
            </Button>

            <ButtonDropDown menuName=<Icon name="sort-down" size="sm" /> right="131px" top="495px">
                <DropDownItem onClick={() => onFilter("OTUs")}>
                    <Checkbox checked={filterOTUs} />
                    OTUs
                </DropDownItem>

                <DropDownItem onClick={() => onFilter("isolates")}>
                    <Checkbox checked={filterIsolates} />
                    Isolates
                </DropDownItem>
            </ButtonDropDown>

            <DropdownButton
                id="download-dropdown"
                title={<PathoscopeDownloadDropdownTitle />}
                pullRight
                style={{ zIndex: 1 }}
            >
                <MenuItem href={`/download/analyses/${analysisId}.csv`}>
                    <Icon name="file-csv" /> CSV
                </MenuItem>
                <MenuItem href={`/download/analyses/${analysisId}.xlsx`}>
                    <Icon name="file-excel" /> Excel
                </MenuItem>
            </DropdownButton>
        </StyledPathoscopeToolbar>
    );
};

export const mapStateToProps = state => {
    const { filterIsolates, filterOTUs, showReads, sortDescending, sortKey } = state.analyses;
    return {
        analysisId: state.analyses.detail.id,
        filterIsolates,
        filterOTUs,
        showReads,
        sortDescending,
        sortKey
    };
};

export const mapDispatchToProps = dispatch => ({
    onCollapse: () => {
        dispatch(collapseAnalysis());
    },

    onFilter: key => {
        if (key !== "OTUs" && key !== "isolates") {
            key = "";
        }

        dispatch(setPathoscopeFilter(key));
    },

    onSetSortKey: key => {
        dispatch(setAnalysisSortKey(key));
    },

    onToggleSortDescending: () => {
        dispatch(togglePathoscopeSortDescending());
    },

    onToggleShowReads: () => {
        dispatch(toggleShowPathoscopeReads());
    }
});

export default connect(mapStateToProps, mapDispatchToProps)(PathoscopeToolbar);
