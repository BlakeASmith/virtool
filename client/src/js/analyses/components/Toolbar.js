import { push } from "connected-react-router";
import React from "react";
import { connect } from "react-redux";
import { FormControl, FormGroup, InputGroup } from "react-bootstrap";
import { Button, Icon } from "../../base";
import { getCanModify } from "../../samples/selectors";
import { findAnalyses } from "../actions";

export const AnalysesToolbar = ({ canModify, onFind, onShowCreate, page, sampleId, term }) => (
    <div className="toolbar">
        <FormGroup>
            <InputGroup>
                <InputGroup.Addon>
                    <Icon name="search" />
                </InputGroup.Addon>
                <FormControl
                    type="text"
                    value={term}
                    onChange={e => onFind(sampleId, e.target.value, page)}
                    placeholder="User or reference"
                />
            </InputGroup>
        </FormGroup>
        <Button
            icon="plus-square fa-fw"
            tip="New Analysis"
            bsStyle="primary"
            onClick={() => onShowCreate(sampleId)}
            disabled={!canModify}
        />
    </div>
);

const mapStateToProps = state => ({
    sampleId: state.samples.detail.id,
    term: state.analyses.term,
    canModify: getCanModify(state)
});

const mapDispatchToProps = dispatch => ({
    onShowCreate: sampleId => {
        dispatch(push({ ...window.location, state: { createAnalysis: [sampleId] } }));
    },
    onFind: (sampleId, term, page) => {
        dispatch(findAnalyses(sampleId, term, page));
    }
});

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(AnalysesToolbar);
