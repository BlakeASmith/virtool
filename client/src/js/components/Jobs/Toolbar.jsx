/**
 * @license
 * The MIT License (MIT)
 * Copyright 2015 Government of Canada
 *
 * @author
 * Ian Boyes
 *
 * @exports JobsToolbar
 */

"use strict";

import React from "react";
import ReactDOM from "react-dom";
import { map } from "lodash";
import { InputGroup, FormGroup, FormControl, Dropdown, MenuItem } from "react-bootstrap";

import Icon from "virtool/js/components/Base/Icon.jsx";
import Flex from "virtool/js/components/Base/Flex.jsx";
import PushButton from "virtool/js/components/Base/PushButton.jsx";

/**
 * A form-based component used to filter the documents presented in JobsTable component.
 *
 * @class
 */
var JobsToolbar = React.createClass({

    getInitialState: function () {
        return {
            task: "",
            username: "",
            pendingRemove: false
        };
    },

    componentDidMount: function () {
        // Focus on the first (task) form field when the component has mounted.
        ReactDOM.findDOMNode(this.refs.find).focus();
    },

    handleSelect: function (eventKey) {

        var toRemove;

        if (eventKey === "removeComplete") {
            toRemove = map(dispatcher.db.jobs.find({
                state: "complete"
            }), "_id");
        }

        if (eventKey === "removeFailed") {
            toRemove = map(dispatcher.db.jobs.find({$or: [
                {state: "error"},
                {state: "cancelled"}
            ]}), "_id");
        }

        if (toRemove) {
            this.clearRemove(toRemove);
        }
    },

    clear: function () {
        var toRemove = map(dispatcher.db.jobs.find({$or: [
            {state: "complete"},
            {state: "error"},
            {state: "cancelled"}
        ]}), "_id");

        this.clearRemove(toRemove);
    },

    clearRemove: function (toRemove) {
        if (toRemove.length > 0) {
            this.setState({pendingRemove: true}, function () {
                dispatcher.db.jobs.request("remove_job", {_id: toRemove})
                    .success(function () {
                        this.setState({pendingRemove: false});
                    }, this)
                    .failure(function () {
                        this.setState({pendingRemove: false});
                    }, this)
            });
        }
    },

    render: function () {

        var removalDropdown;

        if (this.props.canRemove) {
            removalDropdown = (
                <Flex.Item pad>
                    <Dropdown id="job-clear-dropdown" onSelect={this.handleSelect} className="split-dropdown" pullRight>
                        <PushButton onClick={this.clear} tip="Clear Finished">
                            <Icon name="remove" pending={this.state.pendingRemove} />
                        </PushButton>
                        <Dropdown.Toggle />
                        <Dropdown.Menu>
                            <MenuItem eventKey="removeFailed">Failed</MenuItem>
                            <MenuItem eventKey="removeComplete">Complete</MenuItem>
                        </Dropdown.Menu>
                    </Dropdown>
                </Flex.Item>
            );
        }

        return (
            <Flex>
                <Flex.Item grow={1}>
                    <FormGroup>
                        <InputGroup>
                            <InputGroup.Addon>
                                <Icon name="search" /> Find
                            </InputGroup.Addon>
                            <FormControl
                                ref="find"
                                name="find"
                                onChange={this.props.setFindTerm}
                                value={this.props.findTerm}
                            />
                        </InputGroup>
                    </FormGroup>
                </Flex.Item>

                <Flex.Item pad>
                    <PushButton onClick={this.props.changeDirection} tip="Sort Direction">
                        <Icon name={this.props.sortDescending ? "sort-desc": "sort-asc"} />
                    </PushButton>
                </Flex.Item>

                {removalDropdown}
            </Flex>
        );
    }
});

module.exports = JobsToolbar;
