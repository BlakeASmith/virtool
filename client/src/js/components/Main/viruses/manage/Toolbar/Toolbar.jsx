/**
 * @license
 * The MIT License (MIT)
 * Copyright 2015 Government of Canada
 *
 * @author
 * Ian Boyes
 *
 * @exports VirusToolbar
 */

'use strict';

var React = require('react');
var Dropdown = require('react-bootstrap/lib/Dropdown');
var MenuItem = require('react-bootstrap/lib/MenuItem');

var Icon = require('virtool/js/components/Base/Icon.jsx');
var Flex = require('virtool/js/components/Base/Flex.jsx');
var PushButton = require('virtool/js/components/Base/PushButton.jsx');

var Add = require('./Add.jsx');
var Export = require('./Export.jsx');
var Import = require('./Import.jsx');

/**
 * A toolbar component rendered at the top of the virus manager table. Allows searching of viruses by name and
 * abbreviation. Includes a button for creating a new virus.
 */
var VirusToolbar = React.createClass({

    propTypes: {
        onChange: React.PropTypes.func
    },

    getInitialState: function () {
        // The state showAdd is true when the modal should be visible.
        return {
            showAdd: false,
            showExport: false,
            showImport: false,

            flaggedOnly: false,

            canAdd: dispatcher.user.permissions.add_virus,
            canModify: dispatcher.user.permissions.modify_virus
        };
    },

    componentDidMount: function () {
        // Focus on the input field when the component is ready.
        this.refs.input.focus();

        dispatcher.user.on('change', this.onUserChange);
    },

    componentWillUnmount: function () {
        dispatcher.user.off('change', this.onUserChange);
    },

    onUserChange: function () {
        this.setState({
            canAdd: dispatcher.user.permissions.add_virus,
            canModify: dispatcher.user.permissions.modify_virus
        });
    },

    /**
     * Updates the function used to filter virus documents. Triggered by a change in the search input field.
     *
     * @func
     */
    handleChange: function () {
        var re = new RegExp(this.refs.input.value, 'i');

        var filterFunction = function (document) {
            return (
                (re.test(document.name) || re.test(document.abbreviation)) &&
                (!this.state.flaggedOnly || (this.state.flaggedOnly && document.modified))
            );
        }.bind(this);

        this.props.onChange(filterFunction);
    },

    /**
     * Changes state to show the add or export modal form. Triggered by clicking the a menu item.
     *
     * @param event {object} - the select event.
     * @param eventKey {number} - the event key.
     * @func
     */
    handleSelect: function (event, eventKey) {
        switch (eventKey) {

            case 1:
                this.setState({showAdd: true});
                break;

            case 2:
                this.setState({showExport: true});
                break;

            case 3:
                this.setState({showImport: true});
                break;
        }
    },

    toggleFlaggedOnly: function () {
        this.setState({
            flaggedOnly: !this.state.flaggedOnly
        }, this.handleChange);
    },

    /**
     * Changes state to hide the add virus modal form. Triggered as the onHide prop for the modal.
     */
    hideModals: function () {
        this.setState(this.getInitialState(), function () {
            document.getElementById('virus-menu-dropdown').blur();
        });
    },

    render: function () {

        var menu;

        if (this.state.canAdd || this.state.canModify) {
            menu = (
                <Dropdown id='virus-menu-dropdown' pullRight onSelect={this.handleSelect}>
                    <Dropdown.Toggle noCaret ref='menuButton'>
                        <Icon name='menu' />
                    </Dropdown.Toggle>
                    <Dropdown.Menu>
                        <MenuItem eventKey={1} disabled={!this.state.canAdd}>
                            <Icon name='new-entry' /> New
                        </MenuItem>
                        <MenuItem eventKey={2} disabled={!this.state.canModify}>
                            <Icon name='export' /> Export
                        </MenuItem>
                        <MenuItem eventKey={3} disabled={!this.state.canAdd}>
                            <Icon name='new-entry' /> Import
                        </MenuItem>
                    </Dropdown.Menu>
                </Dropdown>
            );
        }

        return (
            <div style={{marginBottom: '15px'}}>
                <Flex>
                    <Flex.Item grow={2}>
                        <div className='input-group'>
                            <span id='find-addon' className='input-group-addon'>
                                <Icon name='search' /> Find
                            </span>
                            <input
                                ref='input'
                                aria-describedby='find-addon'
                                className='form-control'
                                type='text'
                                placeholder='Name or Abbreviation'
                                onChange={this.handleChange}
                            />
                        </div>
                    </Flex.Item>

                    <Flex.Item pad>
                        <PushButton onClick={this.toggleFlaggedOnly} active={this.state.flaggedOnly}>
                            <Icon name='flag' bsStyle='warning' />
                        </PushButton>
                    </Flex.Item>

                    <Flex.Item pad>
                        {menu}
                    </Flex.Item>
                </Flex>

                <Add
                    show={this.state.showAdd}
                    onHide={this.hideModals}
                />

                <Export
                    show={this.state.showExport}
                    onHide={this.hideModals}
                />

                <Import
                    show={this.state.showImport}
                    enableHiding={this.enableHiding}
                    disableHiding={this.disableHiding}
                    onHide={this.hideModals}
                />

            </div>
        );
    }

});

module.exports = VirusToolbar;