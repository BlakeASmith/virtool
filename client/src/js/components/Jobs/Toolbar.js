/**
 * @license
 * The MIT License (MIT)
 * Copyright 2015 Government of Canada
 *
 * @author
 * Ian Boyes
 *
 * @exports JobsToolbar
 */

import React from "react";
import {InputGroup, FormGroup, FormControl, Dropdown, MenuItem} from "react-bootstrap";
import { Icon, Flex, FlexItem, Button } from "virtool/js/components/Base";

export default class JobsToolbar extends React.Component {

    constructor (props) {
        super(props);

        this.state = {
            task: "",
            username: "",
            pendingRemove: false
        };
    }

    static propTypes = {
        findTerm: React.PropTypes.string,
        setFindTerm: React.PropTypes.func,

        sortDescending: React.PropTypes.bool,
        changeDirection: React.PropTypes.func,

        canModify: React.PropTypes.bool,
        canRemove: React.PropTypes.bool
    };

    componentDidMount () {
        this.findNode.focus();
    }

    handleSelect = (eventKey) => {

        let toRemove;

        if (eventKey === "removeComplete") {
            toRemove = dispatcher.db.jobs.find({ state: "complete" }).map(d => d["_id"]);
        }

        if (eventKey === "removeFailed") {
            toRemove = dispatcher.db.jobs.find({$or: [
                {state: "error"},
                {state: "cancelled"}
            ]}).map(d => d["_id"]);
        }

        if (toRemove) {
            this.clearRemove(toRemove);
        }
    };

    clear = () => {
        const toRemove = dispatcher.db.jobs.find({$or: [
            {state: "complete"},
            {state: "error"},
            {state: "cancelled"}
        ]}).map(d => d["_id"]);

        this.clearRemove(toRemove);
    };

    clearRemove = (toRemove) => {
        if (toRemove.length > 0) {
            this.setState({pendingRemove: true}, () => {
                dispatcher.db.jobs.request("remove_job", {_id: toRemove})
                    .success(() => {
                        this.setState({pendingRemove: false});
                    })
                    .failure(() => {
                        this.setState({pendingRemove: false});
                    })
            });
        }
    };

    render () {

        let removalDropdown;

        if (this.props.canRemove) {
            removalDropdown = (
                <FlexItem pad>
                    <Dropdown id="job-clear-dropdown" onSelect={this.handleSelect} className="split-dropdown" pullRight>
                        <Button onClick={this.clear} tip="Clear Finished">
                            <Icon name="remove" pending={this.state.pendingRemove} />
                        </Button>
                        <Dropdown.Toggle />
                        <Dropdown.Menu>
                            <MenuItem eventKey="removeFailed">Failed</MenuItem>
                            <MenuItem eventKey="removeComplete">Complete</MenuItem>
                        </Dropdown.Menu>
                    </Dropdown>
                </FlexItem>
            );
        }

        return (
            <Flex>
                <FlexItem grow={1}>
                    <FormGroup>
                        <InputGroup>
                            <InputGroup.Addon>
                                <Icon name="search" /> Find
                            </InputGroup.Addon>
                            <FormControl
                                name="find"
                                inputRef={(node) => this.findNode = node}
                                onChange={this.props.setFindTerm}
                                value={this.props.findTerm}
                            />
                        </InputGroup>
                    </FormGroup>
                </FlexItem>

                <FlexItem pad>
                    <Button onClick={this.props.changeDirection} tip="Sort Direction">
                        <Icon name={this.props.sortDescending ? "sort-desc": "sort-asc"} />
                    </Button>
                </FlexItem>

                {removalDropdown}
            </Flex>
        );
    }
}
