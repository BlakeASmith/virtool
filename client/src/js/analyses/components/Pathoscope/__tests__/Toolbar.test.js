import {
    COLLAPSE_ANALYSIS,
    SET_ANALYSIS_SORT_KEY,
    SET_PATHOSCOPE_FILTER,
    TOGGLE_SHOW_PATHOSCOPE_READS,
    TOGGLE_SORT_PATHOSCOPE_DESCENDING
} from "../../../../app/actionTypes";
import { PathoscopeDownloadDropdownTitle, PathoscopeToolbar, mapDispatchToProps, mapStateToProps } from "../Toolbar";

describe("<PathoscopeDownloadDropdownTitle />", () => {
    const wrapper = shallow(<PathoscopeDownloadDropdownTitle />);
    expect(wrapper).toMatchSnapshot();
});

describe("<Toolbar />", () => {
    let props;

    beforeEach(() => {
        props = {
            analysisId: "foo",
            filterIsolates: true,
            filterOTUs: true,
            showReads: false,
            sortDescending: true,
            sortKey: "coverage",
            onCollapse: jest.fn(),
            onFilter: jest.fn(),
            onSetSortKey: jest.fn(),
            onToggleShowReads: jest.fn(),
            onToggleSortDescending: jest.fn()
        };
    });

    it("should render", () => {
        const wrapper = shallow(<PathoscopeToolbar {...props} />);
        expect(wrapper).toMatchSnapshot();
    });

    it("should render when [filterIsolates=false]", () => {
        props.filterIsolates = false;
        const wrapper = shallow(<PathoscopeToolbar {...props} />);
        expect(wrapper).toMatchSnapshot();
    });

    it("should render when [filterOTUs=false]", () => {
        props.filterOTUs = false;
        const wrapper = shallow(<PathoscopeToolbar {...props} />);
        expect(wrapper).toMatchSnapshot();
    });

    it("should render when [filterOTUs=false] and [filterIsolates=false]", () => {
        props.filterOTUs = false;
        props.filterIsolates = false;
        const wrapper = shallow(<PathoscopeToolbar {...props} />);
        expect(wrapper).toMatchSnapshot();
    });

    it("should render when [showReads=true]", () => {
        props.filterOTUs = false;
        const wrapper = shallow(<PathoscopeToolbar {...props} />);
        expect(wrapper).toMatchSnapshot();
    });

    it("should render when [sortDescending=false]", () => {
        props.filterOTUs = false;
        const wrapper = shallow(<PathoscopeToolbar {...props} />);
        expect(wrapper).toMatchSnapshot();
    });

    it("should render when [sortKey='weight']", () => {
        props.sortKey = "weight";
        const wrapper = shallow(<PathoscopeToolbar {...props} />);
        expect(wrapper).toMatchSnapshot();
    });

    it("should call onCollapse() when corresponding button clicked", () => {
        const wrapper = shallow(<PathoscopeToolbar {...props} />);
        expect(props.onCollapse).not.toHaveBeenCalled();
        wrapper
            .find("Button")
            .at(1)
            .simulate("click");
        expect(props.onCollapse).toHaveBeenCalled();
    });

    it("should call onFilter() when filter selected", () => {
        const wrapper = shallow(<PathoscopeToolbar {...props} />);
        expect(props.onFilter).not.toHaveBeenCalled();
        wrapper
            .find("#pathoscope-filter-dropdown")
            .props()
            .onSelect("OTUs");
        expect(props.onFilter).toHaveBeenCalledWith("OTUs");
    });

    it("should call onFilter() when filter button clicked", () => {
        const wrapper = shallow(<PathoscopeToolbar {...props} />);
        expect(props.onFilter).not.toHaveBeenCalled();
        wrapper
            .find("Button")
            .at(3)
            .simulate("click");
        expect(props.onFilter).toHaveBeenCalledWith();
    });

    it("should call onSetSortKey() when sort key selected", () => {
        const wrapper = shallow(<PathoscopeToolbar {...props} />);
        const e = {
            target: {
                value: "pi"
            }
        };
        expect(props.onSetSortKey).not.toHaveBeenCalled();
        wrapper.find("FormControl").simulate("change", e);
        expect(props.onSetSortKey).toHaveBeenCalledWith("pi");
    });

    it("should call onToggleShowReads() when corresponding button clicked", () => {
        const wrapper = shallow(<PathoscopeToolbar {...props} />);
        expect(props.onToggleShowReads).not.toHaveBeenCalled();
        wrapper
            .find("Button")
            .at(2)
            .simulate("click");
        expect(props.onToggleShowReads).toHaveBeenCalled();
    });

    it("should call onToggleSortDescending() when corresponding button clicked", () => {
        const wrapper = shallow(<PathoscopeToolbar {...props} />);
        expect(props.onToggleSortDescending).not.toHaveBeenCalled();
        wrapper
            .find("Button")
            .at(0)
            .simulate("click");
        expect(props.onToggleSortDescending).toHaveBeenCalled();
    });
});

describe("mapStateToProps()", () => {
    it("should return props", () => {
        const state = {
            analyses: {
                detail: {
                    id: "foo"
                },
                filterIsolates: false,
                filterOTUs: false,
                showReads: false,
                sortDescending: false,
                sortKey: "pi"
            }
        };
        const props = mapStateToProps(state);
        expect(props).toEqual({
            analysisId: "foo",
            filterIsolates: false,
            filterOTUs: false,
            showReads: false,
            sortDescending: false,
            sortKey: "pi"
        });
    });
});

describe("mapDispatchToProps()", () => {
    let dispatch;
    let props;

    beforeEach(() => {
        dispatch = jest.fn();
        props = mapDispatchToProps(dispatch);
    });

    it("should return onCollapse() in props", () => {
        props.onCollapse();
        expect(dispatch).toHaveBeenCalledWith({
            type: COLLAPSE_ANALYSIS
        });
    });

    it.each(["OTUs", "isolates"])("should return onFilter() in props that works when [key=%p]", key => {
        props.onFilter(key);
        expect(dispatch).toHaveBeenCalledWith({
            type: SET_PATHOSCOPE_FILTER,
            key
        });
    });

    it("should return onFilter() in props that works with non-specific key", () => {
        props.onFilter("foo");
        expect(dispatch).toHaveBeenCalledWith({
            type: SET_PATHOSCOPE_FILTER,
            key: ""
        });
    });

    it("should return onSetSortKey() in props", () => {
        props.onSetSortKey("foo");
        expect(dispatch).toHaveBeenCalledWith({
            type: SET_ANALYSIS_SORT_KEY,
            sortKey: "foo"
        });
    });

    it("should return onToggleSortDescending() in props", () => {
        props.onToggleSortDescending();
        expect(dispatch).toHaveBeenCalledWith({
            type: TOGGLE_SORT_PATHOSCOPE_DESCENDING
        });
    });

    it("should return onToggleShowReads() in props", () => {
        props.onToggleShowReads();
        expect(dispatch).toHaveBeenCalledWith({
            type: TOGGLE_SHOW_PATHOSCOPE_READS
        });
    });
});
