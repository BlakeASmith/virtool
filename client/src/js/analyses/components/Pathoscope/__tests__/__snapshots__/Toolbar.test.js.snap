// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Toolbar /> renders correctly 1`] = `
<StyledComponent
  detail={
    Object {
      "id": "foobar",
    }
  }
  filterIsolates={false}
  filterOTUs={false}
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "Toolbar__StyledPathoscopeToolbar-sc-1p9czx-0",
        "isStatic": true,
        "rules": Array [
          "margin-bottom:10px !important;",
        ],
      },
      "displayName": "Toolbar__StyledPathoscopeToolbar",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "Toolbar__StyledPathoscopeToolbar-sc-1p9czx-0",
      "target": [Function],
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  onCollapse={[Function]}
  onFilter={[Function]}
  onSetSortKey={[Function]}
  onToggleShowMedian={[Function]}
  onToggleShowReads={[Function]}
  onToggleSortDescending={[Function]}
  showMedian={false}
  showReads={false}
  sortDescending={false}
  sortKey=""
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
/>
`;

exports[`<Toolbar /> renders correctly 2`] = `
<StyledComponent
  detail={
    Object {
      "id": "foobar",
    }
  }
  filterIsolates={false}
  filterOTUs={false}
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "Toolbar__StyledPathoscopeToolbar-sc-1p9czx-0",
        "isStatic": true,
        "rules": Array [
          "margin-bottom:10px !important;",
        ],
      },
      "displayName": "Toolbar__StyledPathoscopeToolbar",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "Toolbar__StyledPathoscopeToolbar-sc-1p9czx-0",
      "target": [Function],
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  onCollapse={[Function]}
  onFilter={[Function]}
  onSetSortKey={[Function]}
  onToggleShowMedian={[Function]}
  onToggleShowReads={[Function]}
  onToggleSortDescending={[Function]}
  showMedian={false}
  showReads={false}
  sortDescending={true}
  sortKey=""
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
/>
`;
